client/account-recovery/reset-password-confirm-form/index.jsx

```diff
@@ -29,8 +67,18 @@ class ResetPasswordConfirmForm extends Component {
 					<FormLabel className="reset-password-confirm-form__text-input-label" htmlFor="password">
 						{ translate( 'New password' ) }
 					</FormLabel>
-					<FormPasswordInput className="reset-password-confirm-form__password-input-field" id="password" autoFocus />
-					<FormButton className="reset-password-confirm-form__button generate-password-button" isPrimary={ false }>
+					<FormPasswordInput
+						className="reset-password-confirm-form__password-input-field"
+						id="password"
+						onChange={ this.updateNewPassword }
+						value={ newPassword }
+						autoFocus
+					/>
+					<FormButton
+						className="reset-password-confirm-form__button generate-password-button"
+						type="button"
```

Comment:
Yes, this is intentional. This button is for generating a strong password for users, so it shouldn't submit the form.